pkg/handlers/license.go

```diff
@@ -173,20 +176,76 @@ func (h *Handler) GetLicense(w http.ResponseWriter, r *http.Request) {
 	}
 
 	appSlug := mux.Vars(r)["appSlug"]
-	foundApp, err := store.GetStore().GetAppFromSlug(appSlug)
-	if err != nil {
-		getLicenseResponse.Error = "failed to get app from slug"
-		logger.Error(errors.Wrap(err, getLicenseResponse.Error))
-		JSON(w, http.StatusInternalServerError, getLicenseResponse)
-		return
-	}
+	license := new(kotsv1beta1.License)
+	foundApp := new(apptypes.App)
+
+	isHelmManaged := os.Getenv("IS_HELM_MANAGED")
+	if isHelmManaged == "true" {
+		appCache := getHelmAppCache()
+		app := appCache[appSlug].Application
+		foundApp = &apptypes.App{ID: app.ID, Slug: app.Slug, Name: app.Name}
+		apiEndpoint := os.Getenv("REPLICATED_API_ENDPOINT")
+
+		// get license
+		req, err := util.NewRequest(http.MethodGet, fmt.Sprintf("%s/license", apiEndpoint), nil)
+		if err != nil {
+			getLicenseResponse.Error = "failed to create http request"
+			logger.Error(errors.Wrap(err, getLicenseResponse.Error))
+			JSON(w, http.StatusInternalServerError, getLicenseResponse)
+			return
+		}
+		var licId string
+		if appCache[appSlug].Values["replicated"] != nil && appCache[appSlug].Values["replicated"].(map[string]interface{})["license_id"] != nil {
+			licId = appCache[appSlug].Values["replicated"].(map[string]interface{})["license_id"].(string)
+		}
+		if licId == "" {
+			getLicenseResponse.Error = "replicated license id not present in values"
+			logger.Error(errors.Wrap(err, getLicenseResponse.Error))
+			JSON(w, http.StatusInternalServerError, getLicenseResponse)
+			return
+		}
+		req.SetBasicAuth(licId, licId)
 
-	license, err := store.GetStore().GetLatestLicenseForApp(foundApp.ID)
-	if err != nil {
-		getLicenseResponse.Error = "failed to get license for app"
-		logger.Error(errors.Wrap(err, getLicenseResponse.Error))
-		JSON(w, http.StatusInternalServerError, getLicenseResponse)
-		return
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			getLicenseResponse.Error = "failed to perform http request"
+			logger.Error(errors.Wrap(err, getLicenseResponse.Error))
+			JSON(w, http.StatusInternalServerError, getLicenseResponse)
+			return
+		}
+
+		defer resp.Body.Close()
+		responseBody, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			getLicenseResponse.Error = "failed to read response body"
+			logger.Error(errors.Wrap(err, getLicenseResponse.Error))
+			JSON(w, http.StatusInternalServerError, getLicenseResponse)
+			return
+		}
+
+		license, err = pull.ParseLicenseFromBytes(responseBody)
```

Comment:
i'd suggest using `kotsutil`
```suggestion
		license, err = kotsuitl.LoadLicenseFromBytes(responseBody)
```